blueprint/src/loops.tex

```diff
@@ -155,33 +166,41 @@ \section{Preliminaries}
 
 \begin{proof}
   \leanok
-  \uses{lem:refined_caratheodory}
-  \Cref{lem:refined_caratheodory} gives affinely independent points
-  $p_0, \dots, p_k$ in $P$ such that $x$ is the convex combination of
-  these points with positive weights $w_i$ (here $k \ge 0$).
-  In particular, the family of vectors $p_i - p_0$, $1 ≤ i ≤ k$ is free.
-  So we can complete it to a basis by appending vectors
-  $f_{k+1}, \dots, f_d$.
-  We set $p_i = p_0 + εf_i$ for $i \ge k+1$.
-  This belongs to $P$ for $ε > 0$ small enough since $p_0$ is in $P$
-  and $P$ is open.
-  Note that the extend family of points is still affinely independent
-  since the vectors $p_i - p_0$ are non-zero multiple of our basis
-  elements.
-
-  The next idea is to expand the simplex spanned by the $p_i$'s around its
-  center of mass $b = 1/(d+1)\sum p_i$. So we set:
+  \uses{lem:interior_chab}
+  It follows from \Cref{lem:interior_chab} that we need only show that $E$
+  has an affine basis $b$ of points belonging to $P$ such that $x$ lies in
+  the interior of the convex hull of $b$.
+
+  Carathéodory's lemma \Cref{lem:caratheodory} gives affinely independent
+  points $p_0, \dots, p_k$ in $P$ such that $x$ belongs to their convex
+  hull. Since $P$ is open, we may extend $p_i$ to an affine basis
+  \[
+    \hat b = \{p_0, \ldots, p_d\},
+  \]
+  where all points still belong to $P$.
+  Note that $x$ belongs to the convex hull of $\hat b$.
+
+  Now let $c$ be a point in the interior of the convex hull of $\hat b$
+  (e.g., the centroid) and for each $\epsilon > 0$, consider the homothety
+  \[
+    h_\epsilon : E \to E,
+  \]
+  which dilates about $c$ by a scale of $1 + \epsilon$.
+
+  Since $\hat b$ is finite and contained in $P$, and $P$ is open, there exists
+  $\epsilon > 0$ such that
   \[
-    p'_i = b + (1 + ε)(p_i - b).
+    h_\epsilon (\hat b) \subseteq P.
   \]
-  Those point are all in $P$ for $ε$ small enough because $P$ is open,
-  and there are still affinely independent since they are the image of
-  our original points by a homothety.
-  Then we have:
+  Since $c$ belongs to the interior
+  of the convex hull of $\hat b$ and since $\epsilon > 0$, we have:
   \[
-    x = \sum_i w_i p_i = \sum_i \frac{w_i + ε/(d+1)}{1 + ε} p'_i
+    \Conv(\hat b) \subseteq \IntConv (h_\epsilon (\hat b))
   \]
-  where all coefficients are positive
+  where $\Conv$ and $\IntConv$ denote the convex hull and its interior
+  respectively.
```

Comment:
Could you write more details here?

Reply:
Details added modulo one new lemma whose proof I will add in a few moments.

Reply:
Got sidetracked this afternoon but have just added the missing proof now.